android/src/main/AndroidManifest.xml

```diff
@@ -10,6 +10,8 @@
         android:maxSdkVersion="29" />
     <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
     <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_SPECIAL_USE"/>
```

Comment:
https://developer.android.com/about/versions/14/changes/fgs-types-required says:

> In addition to declaring the FOREGROUND_SERVICE_TYPE_SPECIAL_USE foreground service type, developers should declare use cases in the manifest. To do so, they specify the <property> element within the <service> element. These values and corresponding use cases are reviewed when you submit your app in the Google Play Console. The use cases you provide are free-form, and you should make sure to provide enough information to let the reviewer see why you need to use the specialUse type.

Will this get rejected by the play store without a reason?

Reply:
Sounds like we should use https://developer.android.com/about/versions/14/changes/fgs-types-required#system-exempted given it specifically mentions VPN apps as a use-case.